src/actions/actionCreators.js

```diff
@@ -1,569 +1,28 @@
-    var target = testItem;
-    high = testItem;
-    targetFound = true;
-    isRunning = false
-  }
-  return {
-    type: 'LINEAR_SEARCH',
-    searchArray: searchArray,
-    searchNumber: searchNumber,
-    low: low,
-    high: high,
-    testItem: testItem,
-    index: testItem,
-    iterations: iterations,
-    isRunning: isRunning,
-    targetFound: targetFound,
-    target: target
-  }
-}
-
-
-export function binarySearch(searchArray, searchNumber, low, high, iterations) {
-
-    iterations += 1;
-    var isRunning = true;
-    var targetFound = false;
-
-    var mid = Math.floor((low + high) / 2);
-
-    var testItem = mid;
-
-    if (searchArray[testItem] !== searchNumber && isRunning) {
-      searchNumber < searchArray[testItem] ? high = mid : low = mid;
-      mid = Math.floor((low + high) / 2);
-      testItem = mid;
-
-    } else {
-      testItem = mid;
-      var target = mid;
-      targetFound = true;
-      isRunning = false;
-    }
-    return {
-      type: 'BINARY_SEARCH',
-      searchArray: searchArray,
-      searchNumber: searchNumber,
-      low: low,
-      high: high,
-      mid: mid,
-      testItem: testItem,
-      index: mid,
-      iterations: iterations,
-      isRunning: isRunning,
-      targetFound: targetFound,
-      target: target
-    }
-}
-
-export function changeSpeed(speed, ...timeouts) {
-  // unkown number of timeouts cleared, since different for search/sort algorithm pages
-  // pass timeouts spread, then iterate over each and clear
-  for (var i = 0; i < timeouts.length; i++) {
-    clearTimeout(timeouts[i]);
-  }
-  return {
-    type: 'CHANGE_SPEED',
-    speed
-  }
-}
+import * as  actionBaseSet from './actionBaseSet';
+import * as  actionBinarySearch from './actionBinarySearch';
+import * as  actionBreadthFirstSearch from './actionBreadthFirstSearch';
+import * as  actionLinearSearch from './actionLinearSearch';
+import * as  actionDepthFirstSearch from './actionDepthFirstSearch';
+import * as  actionBubbleSort from './actionBubbleSort';
+import * as  actionHeapSort from './actionHeapSort';
+import * as  actionInsertionSort from './actionInsertionSort';
+import * as  actionMergeSort from './actionMergeSort';
+import * as  actionQuickSort from './actionQuickSort';
+import * as  actionSelectionSort from './actionSelectionSort';
+
+
+let actionObj = {};
+
+actionObj = Object.assign(actionObj, actionBaseSet);
+actionObj = Object.assign(actionObj, actionBinarySearch);
+actionObj = Object.assign(actionObj, actionBreadthFirstSearch);
+actionObj = Object.assign(actionObj, actionLinearSearch);
+actionObj = Object.assign(actionObj, actionDepthFirstSearch);
+actionObj = Object.assign(actionObj, actionBubbleSort);
+actionObj = Object.assign(actionObj, actionHeapSort);
+actionObj = Object.assign(actionObj, actionInsertionSort);
+actionObj = Object.assign(actionObj, actionMergeSort);
+actionObj = Object.assign(actionObj, actionQuickSort);
+actionObj = Object.assign(actionObj, actionSelectionSort);
+
```

Comment:
Remember, we can do this in a loop to clean up the code more.  This approach was to get it started.  You might want to at least add an issue to the GitHub issues to mark that it should be cleaned up later.